core/src/apps/tezos/helpers.py

```diff
@@ -43,12 +46,23 @@ def validate_full_path(path: list) -> bool:
     Validates derivation path to equal 44'/1729'/a',
     where `a` is an account index from 0 to 1 000 000.
     """
-    if len(path) != 3:
+    length = len(path)
+    print(path)
+    if length < 3 or length > 4:
         return False
     if path[0] != 44 | HARDENED:
         return False
     if path[1] != 1729 | HARDENED:
         return False
-    if path[2] < HARDENED or path[2] > 1000000 | HARDENED:
+    if length == 3 and (path[2] < HARDENED or path[2] > 1000000 | HARDENED):
+        return False
+    if length == 4 and (path[2] != 0 | HARDENED or path[3] < HARDENED or path[3] > 1000000 | HARDENED):
```

Comment:
this line is too long
maybe just fork based on length?
```python
if length == 3:
    if path[2]...:
        return False
elif length == 4:
    if path[2] != 0 | HARDENED:
        return False
    if path[3]...:
        return False
```

Reply:
> our CI does not do this right now, so please make sure to run black over all your python files

Sure, will do.

Reply:
Yeah, makes it more readable. Will fix it ASAP